python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -875,3 +881,37 @@ def prepareCalibratedExposures(self, calExpList, backgroundList=None, skyCorrLis
             # Apply skycorr
             if self.config.doApplySkyCorr:
                 mi -= skyCorr.getImage()
+
+
+def reorderInputRefs(inputRefs, outputKeyOrder, dataIdKey):
+    """Reorder inputRefs per outputKeyOrder
+
+    Any inputRefs which are lists will be resorted per specified key e.g.,
+    detector. Only iterables will be reordered, and values can be of type
+    `lsst.pipe.base.connections.DeferredDatasetRef` or
+    `lsst.daf.butler.core.datasets.ref.DatasetRef`
+
+    Parameters
+    ----------
+    inputRefs : `lsst.pipe.base.connections.InputQuantizedConnection`
+        Input references to be reordered and padded.
+    outputKeyOrder : iterable
+        Iterable of values to be compared with inputRef's dataIdKey.
+        Must be hashable
+    dataIdKey :  `str`
+        dataIdKey in the dataRefs to compare with the outputKeyOrder.
+
+    Returns:
+    --------
+    inputRefs: `lsst.pipe.base.connections.InputQuantizedConnection`
+        Sorted inputRefs.
+    """
+    for key, refs in inputRefs:
+        if isinstance(refs, Iterable):
+            if hasattr(refs[0], "dataId"):
+                inputKeyOrder = [ref.dataId[dataIdKey] for ref in refs]
```

Comment:
This also applies to the function above which has just been moved and therefore might be outside the scope of this ticket.  But the term `key` is used in two different ways here.  One is a dict-like `key` (as in `for key, refs in inputRefs`) which is very pythonic.  The other is a sorting `key` which I guess python `sort` functions also use the term `key` in this way, but in this particular method I find the overloading of the term confusing.
It may be verbose, but maybe call `inputKeyOrder` -> `inputSortKeyOrder`?

Reply:
The plain `key` -->  `connectionName`,
`inputKeyOrder` --> `inputSortKeyOrder`, 
`outputKeyOrder` --> `outputSortKeyOrder`